File: source-connector/src/main/java/sourceconnector/repository/file/filter/FileExtensionFilter.java
package sourceconnector.repository.file.filter;


import offsetmanager.domain.file.FileKey;

import java.util.List;

public class FileExtensionFilter implements FileFilter {
  private final List<String> extensions;

  public FileExtensionFilter(List<String> extensions) {
    if (extensions == null || extensions.isEmpty()) {
      throw new IllegalArgumentException("file extensions cannot be null or empty");
    }
    this.extensions = extensions;
  }

  @Override
  public boolean accept(FileKey fileKey) {
    return extensions.stream().anyMatch(extension -> fileKey.get().endsWith(extension));
  }
}

File: source-connector/src/main/kotlin/sourceconnector/repository/file/filter/FileExtensionFilter.kt
package sourceconnector.repository.file.filter

import offsetmanager.domain.file.FileKey


class FileExtensionFilter(extensions: List<String>) : FileFilter {
  private val extensions: List<String>

  init {
    require(extensions.isNotEmpty()) { "file extensions cannot be null or empty" }
    this.extensions = extensions
  }

  override fun accept(fileKey: FileKey): Boolean {
    return extensions.any { extension: String -> fileKey.get().endsWith(extension) }
  }
}
